tiledb/pod/src/array/attribute/mod.rs

```diff
@@ -30,5 +38,118 @@ pub struct FillData {
     pub nullability: Option<bool>,
 }
 
-#[cfg(any(test, feature = "proptest-strategies"))]
-pub mod strategy;
+/// Identifies an [EnumerationData] which this attribute indexes.
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+pub enum EnumerationRef {
+    /// The name of an enumeration whose data lives elsewhere.
+    Name(String),
+    /// The enumeration is uniquely owned by this attribute.
+    ///
+    /// In typical usage, this is an ephemeral state before the
+    /// attribute is attached to a [SchemaData].
+    OwnedByAttribute(EnumerationData),
+    /// This attribute is owned by a [SchemaData] and is sharing
+    /// its enumeration data.
+    ///
+    /// This is the expected variant for attributes which are
+    /// owned by [SchemaData]. Other variants may be used transiently.
+    /// For example, serialization of this variant writes only the enumeration name,
+    /// and deserialization of the owning [SchemaData] is responsible for
+    /// reconstructing this variant.
+    #[cfg_attr(
+        feature = "serde",
+        serde(
+            rename = "Name",
+            serialize_with = "EnumerationRef::serialize_owned_by_schema",
+            skip_deserializing
+        )
+    )]
+    BorrowedFromSchema(Rc<EnumerationData>),
```

Comment:
> This is also slightly odd given that Borrowed suggests to me that the EnumerationData would have lifetimes, but it owns its data buffers.

Totally fair.  This came about because by default `serde` does not like `Rc`, because it can't preserve the sharing relationships when you deserialize (there is an optional `serde` feature to enable this, but I don't think that's the right thing to do in a library). Initially I just wanted to do `Rc<EnumerationData>` but this `serde` limitation got in the way.  I definitely didn't want a proper Rust borrow either because that would lead to lots of other cruft for not a lot of realized value.  I chose the name `Borrow` to reflect that the Schema is the thing that owns the data to be serialized/deserialized and the attributes in the schema just borrow that.

I'll go with anything for the name; `OwnedBySchema` was something else I considered

Reply:
Resolved by 5115ce07